aspnet/migration/rc1-to-rtm.rst

```diff
@@ -0,0 +1,503 @@
+
+The environment key is set with the ``ASPNETCORE_ENVIRONMENT`` environment variable. ``ASPNET_ENV`` and ``Hosting:Environment`` are still supported, but generate a deprecated message warning.
+
+Hosting service changes
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Dependency injection code that uses ``IApplicationEnvironment`` must now use :dn:iface:`~Microsoft.AspNetCore.Hosting.IHostingEnvironment`. For example, in your ``Startup`` class, change:
+
+.. code-block:: c#
+
+  public Startup(IApplicationEnvironment applicationEnvironment)
+
+To:
+
+.. code-block:: c#
+
+  public Startup(IHostingEnvironment hostingEnvironment)
+
+Kestrel
+-------
+
+Kestrel configuration has changed. `This GitHub announcement <https://github.com/aspnet/Announcements/issues/168>`_ outlines the changes you must make to configure Kestrel if you are not using default settings.
+
+Controller and action results renamed
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following :dn:class:`~Microsoft.AspNetCore.Mvc.Controller` methods have been renamed and moved to :dn:class:`~Microsoft.AspNetCore.Mvc.ControllerBase`:
+
+==================================  ==================
+ASP.NET 5 RC1                       ASP.NET Core 1.0
+==================================  ==================
+HttpUnauthorized                    Unauthorized
+HttpNotFound (and its overloads)    NotFound
+HttpBadRequest (and its overloads)  BadRequest
+==================================  ==================
+
+The following action result types have also been renamed:
+
+=============================================  =============================================
+ASP.NET 5 RC1                                        ASP.NET Core 1.0
+=============================================  =============================================
+Microsoft.AspNet.Mvc.HttpOkObjectResult        Microsoft.AspNetCore.Mvc.OkObjectResult
+Microsoft.AspNet.Mvc.HttpOkResult              Microsoft.AspNetCore.Mvc.OkResult
+Microsoft.AspNet.Mvc.HttpNotFoundObjectResult  Microsoft.AspNetCore.Mvc.NotFoundObjectResult
+Microsoft.AspNet.Mvc.HttpNotFoundResult        Microsoft.AspNetCore.Mvc.NotFoundResult
+Microsoft.AspNet.Mvc.HttpStatusCodeResult      Microsoft.AspNetCore.Mvc.StatusCodeResult
+Microsoft.AspNet.Mvc.HttpUnauthorizedResult    Microsoft.AspNetCore.Mvc.UnauthorizedResult
+=============================================  =============================================
+
+ASP.NET 5 MVC compile views
+---------------------------
+
+To compile views, set the ``preserveCompilationContext`` option in *project.json* to preserve the compilation context, as shown here:
```

Comment:
I don't know that "to preserve the compilation context" will mean much to most people. If we care that they know why `preserveCompilationContext` is the way to compile views then we might want a more detailed description of what it does.